File: projects/m1/023-parity-bits/solutions/GabrieleGabellone/parity_bits.py
#Parity Bits
#this program calculates the parity bit for groups of 8 bits entered by the user using even parity

while True:
    user_bits = str(input("Enter a group of bits (blank to quit): "))
    if len(user_bits) != 8:
        if len(user_bits) == 0: break
        else: 
            print("Error! The group must be 8 bits") 
            continue
    if user_bits.count("1") % 2 == 0: print("The parity bit should be 0")
    else: print("The parity bit should be 1")

File: projects/m1/023-parity-bits/solutions/alessiacaporaso/parity-bits.js
// read strings containing 8 bits until the user enters a blank line.
const bits = prompt('Enter a bit')
const bit = bits.split("")
let message = "";
let calcBit = 0;
if(bit.length < '8' || bit.length > '8'){
  message = "Error!"
} else {
  for (const item of bit) { 
    calcBit += parseInt(item)
  }
  if(calcBit % 2 === 0) {
    message = "The parity bit should be  0"
  } else {
    message = "The parity bit should be  1"
  }
}
alert(message)